Armadillo/src/main/java/com/scribd/armadillo/download/drm/ArmadilloDrmSessionManagerProvider.kt

```diff
@@ -0,0 +1,136 @@
+import com.google.android.exoplayer2.util.Assertions
+import com.google.android.exoplayer2.util.Util
+import com.google.common.collect.UnmodifiableIterator
+import com.google.common.primitives.Ints
+import com.scribd.armadillo.StateStore
+import com.scribd.armadillo.actions.LicenseDrmErrorAction
+import com.scribd.armadillo.actions.LicenseExpirationDetermined
+import com.scribd.armadillo.actions.LicenseExpiredAction
+import com.scribd.armadillo.actions.LicenseKeyIsUsableAction
+import com.scribd.armadillo.time.milliseconds
+import java.util.UUID
+import javax.inject.Inject
+
+/** Forks [DefaultDrmSessionManagerProvider], but changes the [ExoMediaDrm.Provider] instance
+ *
+ * note: if moving to media3, maybe AppManagedProvider may help*/
+internal class ArmadilloDrmSessionManagerProvider @Inject constructor(private val stateStore: StateStore.Modifier) :
+    DrmSessionManagerProvider {
+
+    ////////////////////////////////////////////////////////////////
+    // These fields identical to DefaultDrmSessionManagerProvider //
+
+    private val lock = Any()
+
+    @GuardedBy("lock")
+    private var drmConfiguration: DrmConfiguration? = null
+
+    @GuardedBy("lock")
+    private var manager: DrmSessionManager? = null
+    private val drmHttpDataSourceFactory: DataSource.Factory? = null
+    private val userAgent: String? = null
+
+    // End DefaultDrmSessionManagerProvider fields //
+    /////////////////////////////////////////////////
+
+    companion object {
+        const val TAG = "ArmadilloDrmSMProvider"
+    }
+
+    /**  Identical to DefaultDrmSessionManagerProvider method, */
+    override fun get(mediaItem: MediaItem): DrmSessionManager {
+        Assertions.checkNotNull(mediaItem.localConfiguration)
+        val drmConfiguration = mediaItem.localConfiguration!!.drmConfiguration
+        if (drmConfiguration != null && Util.SDK_INT >= 18) {
+            synchronized(this.lock) {
+                if (!Util.areEqual(drmConfiguration, this.drmConfiguration)) {
+                    this.drmConfiguration = drmConfiguration
+                    this.manager = this.createManager(drmConfiguration)
+                }
+                return Assertions.checkNotNull<DrmSessionManager?>(this.manager)
+            }
+        } else {
+            return DrmSessionManager.DRM_UNSUPPORTED
+        }
+    }
+
+    /** Near identical to DefaultDrmSessionManagerProvider method, except for the indicated line */
+    private fun createManager(drmConfiguration: DrmConfiguration): DrmSessionManager {
+        val dataSourceFactory = this.drmHttpDataSourceFactory
+            ?: DefaultHttpDataSource.Factory().setUserAgent(this.userAgent)
+        val httpDrmCallback = HttpMediaDrmCallback(if (drmConfiguration.licenseUri == null) null else drmConfiguration.licenseUri.toString(), drmConfiguration.forceDefaultLicenseUri, (dataSourceFactory)!!)
+        val var4: UnmodifiableIterator<*> = drmConfiguration.licenseRequestHeaders.entries.iterator()
```

Comment:
Maybe clean up the `DefaultDrmSessionManagerProvider` code a bit, or leave it to better convey what it's copied from? idk